modules/oci-build/image_tool/append_layers.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/match"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/spf13/cobra"
+)
+
+var CommandAppendLayers = cobra.Command{
+	Use:   "append-layers oci-path [path-to-tarball...]",
+	Short: "Appends a tarball or directory to every image in an OCI index.",
+	Args:  cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		oci := args[0]
+		extra := args[1:]
+
+		if len(extra) == 0 {
+			return
+		}
+
+		layers := []v1.Layer{}
+		for _, path := range extra {
+			layers = append(layers, loadLayerFromDirOrTarball(path))
+		}
+
+		appendLayersToAllImages(oci, layers...)
+	},
+}
+
+func loadLayerFromDirOrTarball(path string) v1.Layer {
+	stat, err := os.Stat(path)
+	must("could not open directory or tarball", err)
+
+	var layer v1.Layer
+	if stat.IsDir() {
+		var buf bytes.Buffer
+
+		tw := tar.NewWriter(&buf)
+
+		filepath.Walk(path, func(target string, info fs.FileInfo, err error) error {
+			must("walk error", err)
+
+			header, err := tar.FileInfoHeader(info, info.Name())
+			must("could not create tar header", err)
+
+			name, err := filepath.Rel(path, target)
+			must("could not build relative path", err)
+
+			// Write simplified header, this removes all fields that would cause
+			// the build to be non-reproducible (like modtime for example)
+			err = tw.WriteHeader(&tar.Header{
+				Typeflag: header.Typeflag,
+				Name:     name,
+				Mode:     header.Mode,
+				Linkname: header.Linkname,
+				Size:     header.Size,
```

Comment:
What does the UID and GID default to?
When you ls in the container image, what is the result that you see?

Reply:
ko also does not seem to set this value, so it is probably fine? https://github.com/ko-build/ko/blob/e407275248762120557098982154132dfec9a01f/pkg/build/gobuild.go#L635-L644
Maybe we should also overwrite the Mode like ko to 0555?